### DISCUSSION TITLE
Sentry Init with GlobalHubMode Enabled

### DISCUSSION THREAD

**Discussion by kathapatel** [getsentry/sentry-java#2581] [Q&A] [answered]
Hi Team,

currently I am integrating Sentry with my application, I am initializing SentryOptions bean with my application property. Is there way to initialize Sentry other than
  public static <T extends SentryOptions> void init(
      final @NotNull OptionsConfiguration<T> optionsConfiguration,
      final boolean globalHubMode){}

This way we'll need to hardcode configuration and we won't be able to fetch from application property automatically.
 
can we add init method with below signature?
public static void init(SentryOptions options, boolean globalHubMode)?

How reliable GlobalHubMode is?

Thanks.

**Comment by romtsn** [comment, marked as answer]
hi, what is your application exactly? Is it backend? For backend in general you should not use the global hub mode, because it will mess up the SDK when working in a multi-threaded environment... but @adinauer has more opinions on that probably.

**Reply by adinauer** [reply]
As @romtsn said, `globalHubMode` is intended for Android (and maybe Java Desktop applications), not for backend application. Can you please share more details on why you need `globalHubMode`?

**Reply by kathapatel** [reply]
Got it, Thanks, yes we were configuring for backend services.

**Reply by kathapatel** [reply]
@adinauer checked internal implementation of Sentry Hub and understood globalHubMode will not work in backend service, we were considering globalHubMode as part of optimization.
